sympy/stats/error_prop.py

```diff
@@ -0,0 +1,62 @@
+"""Tools for arithmetic error propogation."""
+from __future__ import print_function, division
+from itertools import repeat
+from sympy import Symbol, symbols, Add, Mul, simplify, Pow, exp
+
+
+def variance_prop(expr, consts=()):
+    """Symbolically propagates variance (sigma^2) for expressions.
+
+    Parameters
+    ==========
+    expr : Expr
+        A sympy expression to compute the variance for.
+    consts : sequence of Symbols, optional
+        Represents symbols that are known constants in the expr,
+        and thus have zero variance.
+
+    Returns
+    =======
+    var_expr : Expr
+        An expression for the total variance of the expr.
+        The variance for the original symbols (e.g. x) have a 'var_'
+        prepended to their name ('var_x').
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, exp
+    >>> from sympy.stats.error_prop import variance_prop
+    >>> x, y = symbols('x y')
+
+    >>> variance_prop(x + y)
+    var_x + var_y
+
+    >>> variance_prop(x * y)
+    var_x*y**2 + var_y*x**2
+
+    >>> variance_prop(exp(2*x))
+    4*var_x*exp(4*x)
+
+    """
+    args = expr.args
+    if len(args) == 0:
+        if isinstance(expr, Symbol) and expr not in consts:
+            signame = 'var_' + expr.name
```

Comment:
I agree using some special object would be better here, if it makes sense. Denoting things by symbol names is generally avoided in SymPy (although it is used in some places, e.g., arbitrary constants in dsolve).